MonsterRace/src/main/java/model/Monsters.java

```diff
@@ -0,0 +1,42 @@
+package model;
+
+public abstract class Monsters {
+
+  protected String name;
+  protected int move;
+
+  public Monsters(String name) {
+    this.name = name;
+    this.move = 0;
+  }
+
+  public boolean isMove(Integer randomNumber) {
+    return false;
+  }
+
+  public void updateResults(Integer randomNumber) {
+  }
+
+  public void setMove(int moveNum){
+    this.move = moveNum;
+  }
+
+  public int getMoveNum(){
+    return this.move;
+  }
+
+  public String getName() {
+    return this.name;
+  }
+
+  @Override
+  public String toString() {
+    StringBuilder builder = new StringBuilder();
+    builder.append(getName()).append(" ").append(getMoveNum());
+    for (int i = 0; i < this.move; i++) {
+      builder.append("-");
+    }
+    builder.append("\n");
+    return builder.toString();
+  }
```

Comment:
`java.lang.Object#toString()` 을 오버라이드할 시에 내부 로직이 지나치게 복잡해지는 것 피하면 좋겠습니다. 
`toString()` 은 객체의 상태를 보여줄 수 있는 용도로 한정해야 합니다. 따라서 필드를 변형 없이 그대로 노출할 수 있는 정도의 오버라이드가 딱 좋고, IDE의 auto-generation에 의존하는 것도 나쁘지 않습니다. 

지금 이 메소드의 용처를 살펴보니 결과값 노출에 사용되고 있는데, 결과값을 만들어줄 메소드를 따로 분리해 내기를 권합니다. 

과거 제가 비슷한 방향의 구현을 했다가 디버깅하는데 세 시간 넘게 소모한 기억이 납니다. 
사이드 이펙트는 언제 어디서 튀어나올 지 모르니 유의합시다.

Reply:
이번 코드에서는 확인하셨다시피, 각 객체들이 `toString()`을 통해 생성해 준 결과값을 컨트롤러가 모아모아 마지막에 보여주는 역할을 수행합니다. toString은 필드만 드러내는 역할을 하고, 자기 객체의 움직임을 만드는 역할을 수행하는 메소드를 클래스 내부에 만든뒤, 이를 외부의 출력담당 메소드/클래스에 던지는 역할을 수행하면 좋을 것이다 라고 이해해도 괜찮을까요?

어제 밤 자기 전에 [Printers instead of getters](https://www.yegor256.com/2016/04/05/printers-instead-of-getters.html) 라는 글을 읽었는데요. 여기서는 `setter()` 형식으로 결과값을 출력하지 말고 별도로 결과값이 print되는 객체를 하나 만들어서 이를 담당 객체에 보내 자기에게 해당하는 값을 찍어서 리턴하도록 하자는 주장인데요. 객체를 죽은 놈으로 만들지 말고 active하게 살려서 사용하자..는 주장인데. 우리의 미션에 맞게 생각해보면, 최종적으로 print되는 객체를 하나 만든 뒤 이를 각 객체를 순회하면서 객체 결과 출력 담당자가 .append() 하는 형식으로 구현해봐도 괜찮을까요? 이런 식으로 객체의 생명력을 중요하게 생각해야 하는 지도 궁금하네요. :)

Reply:
`Printer` 라고 하시기에 `PrintWriter` 같은 건 줄 알고 긴장했는데 다행히 아니군요. ㅎㅎ
네 저도 링크해주신 글이 들고 있는 예제처럼 구현되기를 바랍니다. 그게 더 좋습니다. 

순회해서 `.append()` 하는 것 좋습니다.
리뷰이 분들께 공통 피드백드리고 싶은 내용인데 객체 밖에서 순회하려 하지 말고 객체들의 상태를 다 같이 관리하는 다른 객체를 설계하는 것은 어떨까? 라는 힌트를 드리고 싶네요. 여기까지 ㅎㅎㅎ

Reply:
맞는 지는 모르겠지만, [스테이트 패턴](https://victorydntmd.tistory.com/294) 같은 걸 활용하면 될 지도 모르겠군요. 다음 PR 때 조금 더 고민해보겠습니다.

Reply:
디자인 패턴에 너무 구애받지 마세요. 아직 그럴 시기가 아닙니다.